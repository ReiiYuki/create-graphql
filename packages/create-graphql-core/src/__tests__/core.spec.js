import path from 'path';
import core from '../index';

/**
 * Get a fixture path
 * @param name {string} Name of the file of the fixture
 * @returns {string} The path of the fixture
 */
const getFixturePath = name => path.join(__dirname, `../../fixtures/${name}`);

it('should call getFields of first plugin, and generateTemplate from the second one', async () => {
  const testName = 'mongoose-type';
  const testPath = getFixturePath(testName);

  const opts = {
    plugins: ['mongoose', 'type'],
    name: 'User',
    schema: 'User',
  };

  const templates = await core(testPath, opts);

  expect(templates).toMatchSnapshot();
});
