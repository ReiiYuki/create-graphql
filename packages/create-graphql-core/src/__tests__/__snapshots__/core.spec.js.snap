// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call getFields of first plugin, and generateTemplate from the second one 1`] = `
Array [
  "export default new GraphQLObjectType({
  name: \\"User\\",
  description: 'Represents NAME',
  fields: () => ({
    id: globalIdField('NAME'),
    example: {
      type: GraphQLString,
      description: 'My example field',
      resolve: obj => obj.example
    }
  }),
  interfaces: () => [NodeInterface]
});",
]
`;
